standalone.go

```diff
@@ -53,29 +63,87 @@ func (s *standalone) pick() int {
 	return rand.IntN(len(s.replicas))
 }
 
+func (s *standalone) handleRedirect(ctx context.Context, cmd Completed, result ValkeyResult) ValkeyResult {
+	if !s.enableRedirect {
+		return result
+	}
+
+	if ret, yes := IsValkeyErr(result.Error()); yes {
+		if addr, ok := ret.IsRedirect(); ok {
+			// Use singleflight to ensure only one redirect operation happens at a time
+			if err := s.redirectCall.Do(ctx, func() error {
+				return s.redirectToPrimary(addr)
+			}); err != nil {
+				// If redirect fails, return the original result
+				return result
+			}
+
+			// Execute the command on the updated primary
+			return s.primary.Do(ctx, cmd)
```

Comment:
What if this returns a redirect error again?

Reply:
Do we retry it? @rueian 
Maybe we can take in a configurable retry limit, What do you suggest?